head.go

```diff
@@ -1349,6 +1361,7 @@ func (h *Head) getOrCreateWithID(id, hash uint64, lset labels.Labels) (*memSerie
 
 	h.metrics.series.Inc()
```

Comment:
We're tracking this through `NumSeries()`. Can we just drop the gauge and use gaugeFunc()

Reply:
Will do. Also, I have another PR for `NumSeries()` https://github.com/prometheus/tsdb/pull/656 if that would help in reviewing. Do you want to be in a separate PR or inside this PR itself?